lib/io/hardware_i2c.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "i2c.hpp"
+
+#include <optional>
```

Comment:
already included in `i2c.hpp` - and `types.hpp` is already included too

Reply:
we don't like transitive includes because they can make things intransparent so this is fine imo